animation_nodes/nodes/gpencil/gp_layer_info.py

```diff
@@ -0,0 +1,141 @@
+
+    useFrameList: VectorizedSocket.newProperty()
+
+    def create(self):
+        self.newInput("GPLayer", "Layer", "layer", dataIsModified = True)
+
+        if self.frameType == "ACTIVE":
+            self.newInput("Scene", "Scene", "scene", hide = True)
+
+            self.newOutput("GPFrame", "Frame", "frame")
+            self.newOutput("Integer", "Frame Index", "frameIndex")
+            self.newOutput("Float", "Frame Number", "frameNumber")
+        elif self.frameType == "INDEX":
+            self.newInput(VectorizedSocket("Integer", "useFrameList",
+            ("Frame Index", "frameIndex"), ("Frame Indices", "frameIndices")))
+
+            self.newOutput(VectorizedSocket("GPFrame", "useFrameList",
+            ("Frame", "frame"), ("Frames", "frames")))
+            self.newOutput(VectorizedSocket("Float", "useFrameList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "outframeNumbers")))
+        elif self.frameType == "FRAME":
+            self.newInput(VectorizedSocket("Float", "useFrameList",
+            ("Frame Number", "frameNumber"), ("Frame Numbers", "inframeNumbers")))
+
+            self.newOutput(VectorizedSocket("GPFrame", "useFrameList",
+            ("Frame", "frame"), ("Frames", "frames")))
+            self.newOutput(VectorizedSocket("Integer", "useFrameList",
+            ("Frame Index", "frameIndex"), ("Frame Indices", "outframeIndices")))
+        elif self.frameType == "ALL":
+            self.newOutput("GPFrame List", "Frames", "frames")
+            self.newOutput("Float List", "Frame Numbers", "outframeNumbers")
+        self.newOutput("Text", "Layer Name", "layerName", hide = True)
+        self.newOutput("Text", "Layer Blend Mode", "blendMode", hide = True)
+        self.newOutput("Float", "Layer Opacity", "opacity", hide = True)
+        self.newOutput("Integer", "Layer Pass Index", "passIndex", hide = True)
+
+    def draw(self, layout):
+        layout.prop(self, "frameType", text = "")
+
+    def getExecutionFunctionName(self):
+        if self.frameType == "ACTIVE":
+            return "execute_ActiveFrame"
+        elif self.frameType == "INDEX":
+            if self.useFrameList:
+                return "execute_FrameIndices"
+            else:
+                return "execute_FrameIndex"
+        elif self.frameType == "FRAME":
+            if self.useFrameList:
+                return "execute_FrameNumbers"
+            else:
+                return "execute_FrameNumber"
+        elif self.frameType == "ALL":
+            return "execute_AllFrames"
+
+    def execute_ActiveFrame(self, layer, scene):
+        frames = layer.frames
+        if len(layer.frames) == 0:
+            return GPFrame(), 0, 0, layer.layerName, layer.blendMode, layer.opacity, layer.passIndex
+        frameNumbers = layer.frameNumbers
+        activeIndex = self.getActiveFrameIndex(scene.frame_current, frameNumbers)
+        return frames[activeIndex], activeIndex, frameNumbers[activeIndex], layer.layerName,\
```

Comment:
You can put those in parentheses to avoid the `\` at the end.

Reply:
Okay.